### PR TITLE
Junos provider readd

### PR DESCRIPTION
**Pull request by Qalthos**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
junos_config, etc

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY
Addresses a number of issues within the junos modules

Addresses #21824, #21826, #21827

### CODE REVIEW

File: lib/ansible/module_utils/junos.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
     'password': dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), no_log=True),
     'ssh_keyfile': dict(fallback=(env_fallback, ['ANSIBLE_NET_SSH_KEYFILE']), type='path'),
     'timeout': dict(type='int', default=10),
-    'provider': dict(type='dict'),
+    'provider': dict(type='dict', no_log=True),
```
Comment (gundalow) [inline lib/ansible/module_utils/junos.py]: Not sure if that's needed, it's only the password that he should be masked out, which I believe is done already.

Comment (gundalow) [reply lib/ansible/module_utils/junos.py]: I think this is a regression in the argspec work that went into 2.3 for Network modules, raised as https://github.com/ansible/ansible/issues/21892

File: lib/ansible/modules/network/junos/_junos_template.py [hunk 2]
```diff
@@ -127,9 +129,13 @@ def main():
         transport=dict(default='netconf', choices=['netconf'])
     )
 
+    argument_spec.update(junos_argument_spec)
+
     module = AnsibleModule(argument_spec=argument_spec,
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/_junos_template.py:135]: `Undefined variable 'AnsibleModule'`

File: lib/ansible/modules/network/junos/_junos_template.py [hunk 3]
```diff
@@ -127,9 +129,13 @@ def main():
...
                            supports_check_mode=True)
 
+    check_args(module, warnings)
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/_junos_template.py]: `Undefined variable 'warnings'`

**Review by gundalow — CHANGES_REQUESTED** [review]
Looks like the imports are broken, unless I have't applied your patch correctly locally.

Might be worth running `pylint -E network/junos`

File: lib/ansible/modules/network/junos/_junos_template.py (hunk 2 above)
Comment (Qalthos) [reply lib/ansible/modules/network/junos/_junos_template.py:135]: Fixed in c63f187a3e9e670531b7b3f042cf3c134bedcd93

File: lib/ansible/modules/network/junos/_junos_template.py (hunk 3 above)
Comment (Qalthos) [reply lib/ansible/modules/network/junos/_junos_template.py]: Fixed in c63f187

**Pull request merged**